### ISSUE TITLE
[BUG] VMs created via rancher integration not removable

### ISSUE THREAD

**Issue by johnwc** [harvester/harvester#6047]
**Describe the bug**
We had a long power outage in our lab that caused servers to go offline without proper shutdown. It ended up causing one of our test k8s clusters to get corrupted to a point where we did not want to put time into fixing it vs rebuilding it. We went into rancher and deleted the problematic cluster. All the VMs shutdown and are deleted in harvester except two. Within harvester, one shows as `Off`, but has no `Start` menu option but has a `Force Stop` menu option. Using the `Force Stop` option gives an error that the VM is not running. The other shows status of `Starting`, but menu has `Start`, and `Restart` menu options. Trying to use the `Delete` menu option on either gives us the delete verification prompt along with optional disks to also delete, but after clicking `Delete` on the modal they do nothing, with or without the disks selected. I have put each harvester node in Maintenance mode and rebooted one by one, to attempt to see if it was a node that had it stuck.

How can we delete these VMs from harvester as well as get the cluster deleted from rancher? (Even if it requires manually editing config yaml in harvester's rancher manager to release them, then manual clean up)

**Environment**
 - Harvester ISO version: 1.2.2
 - Underlying Infrastructure: Dell PowerEdge R640:

**Additional context**
Rancher screenshots
![Screenshot 2024-06-20 022822](https://github.com/harvester/harvester/assets/2798441/c29b261a-ee66-4583-b555-fb4f9e5ddaa5)

![Screenshot 2024-06-20 023340](https://github.com/harvester/harvester/assets/2798441/52f5184f-cd88-4ae3-b5a4-ed1c168d7710)

Harvester screenshot
![Screenshot 2024-06-20 023813](https://github.com/harvester/harvester/assets/2798441/05d04090-cf15-4931-bfe4-37cddbcce8cb)

**Comment by w13915984028** [comment]
@johnwc Please try to follow this workaround to delete the related resources.

(normally, one running VM will have `vm, vmi, pod` resources)

https://docs.harvesterhci.io/v1.3/troubleshooting/vm/#workaround

**Comment by johnwc** [comment]
@w13915984028 there were no pods corresponding to the two VMs unfortunately. What I ended up having to do was... 

I stumbled across the path `https://<harvester-vip>/dashboard/c/local/explorer/kubevirt.io.VirtualMachine`, where I saw the two VMs. I deleted them from that interface, and they ended up getting deleted for good. Then I had to force delete the two `cluster.x-k8s.io.machine` objects from rancher's local cluster.
